src/openforms/submissions/models/submission.py

```diff
@@ -708,7 +709,39 @@ def calculate_price(self, save=True) -> None:
 
     @property
     def payment_required(self) -> bool:
-        return self.form.payment_required
+        if not self.form.payment_required:
+            return False
+
+        # test the rules one by one, if relevant
+        price_rules = self.form.formpricelogic_set.all()
+        for rule in price_rules:
```

Comment:
I'm not a fan of this code duplication with `calculate_price` - can we not use `calculate_price` instead and if that returns a non-zero value, payment is required, otherwise it's not?

I think we can remove the `payment_required` check from that function, since there's a circular dependency otherwise.

Reply:
Yes, I agree I didn't want to add duplicate code but I was facing some problems..
I will look again if removing the `payment_required` from the function does not affect any other functionality.